lib/npg_qc/autoqc/role/generic.pm

```diff
@@ -17,10 +16,40 @@ around 'filename_root' => sub {
   return join q[.], $self->$orig(), $self->pp_name || q[unknown];
 };
 
-sub doc2yaml {
+sub massage_for_render {
   my $self = shift;
-  my $doc = $self->doc;
-  return $doc ? Dump($doc) : q[];
+
+  my %table_data = ();
+  if (exists $self->doc->{'QC summary'}) {
+    my $qc_data = $self->doc->{'QC summary'};
```

Comment:
We've checked for existence of the value, but it can potentially be undefined (the same for the meta key). We can do this:
```my $qc_data = $self->doc->{'QC summary'} || {};```